overlord/snapstate/progress.go

```diff
@@ -75,13 +88,8 @@ func (t *taskProgressAdapter) Finished() {
 
 // Write sets the current write progress
 func (t *taskProgressAdapter) Write(p []byte) (n int, err error) {
-	if t.unlocked {
-		t.task.State().Lock()
-		defer t.task.State().Unlock()
-	}
-
 	t.current += float64(len(p))
-	t.task.SetProgress(t.label, int(t.current), int(t.total))
+	t.Set(t.current)
```

Comment:
If this got renamed to eg. `func (t *taskProgressAdapter) Add(much float64)`, then you could just do:
```
func (t *taskProgressAdapter) Write(p []byte) (n int, err error) {
    t.Add(float64(len(p)))
    return len(p), nil
}
```
and not set `t.current` twice (here and in `t.Set()`)

Reply:
We use Set to implement the "progress.Meter" interface.